### PR TITLE
[WIP] Add new VideoLoader class for faster video reading.

### PR DESCRIPTION
**Pull request by IgorSusmelj**
### Changes

- Use new `video_reader` backend from torchvision for faster video loading

### Todo

- [x] Add tests
- [x] Remove old `_video_loader` 
- [ ] Use new loader for faster embedding of videos

### CODE REVIEW

**Comment by IgorSusmelj** [comment]
@philippmwirth, the new video interface is much faster when reading sequential frames. That might not work when using multiple workers. 

How should we change the embedding part of the code? Just switch to 0 or 1 worker in case we can use the new backend?

**Comment by philippmwirth** [comment]
> 
> 
> @philippmwirth, the new video interface is much faster when reading sequential frames. That might not work when using multiple workers.
> 
> How should we change the embedding part of the code? Just switch to 0 or 1 worker in case we can use the new backend?

I would leave it to the user and document it clearly. IMO it's not a good idea to enforce a certain number of workers.

File: lightly/data/_video.py [hunk 1]
```diff
@@ -47,6 +48,117 @@ def _video_loader(path, timestamp, pts_unit='sec'):
...
+
+        else: # fallback on pyav
+            if timestamp == None:
+                timestamp = self.timestamps[self.current_timestamp_idx]
+            frame, _, _ = io.read_video(self.path,
+                                        start_pts=timestamp,
+                                        end_pts=timestamp,
+                                        pts_unit=self.pts_unit)    
+            self.last_timestamp_idx = self.timestamps.index(timestamp)    
+
```
Comment (philippmwirth) [inline lightly/data/_video.py:145]: Nice idea! I have some feedback:
- Be consistent in the way you check for `None` (I would always go with `if x is None` or `if x is not None`).
- A few comments would make reading this part a lot easier.

**Comment by IgorSusmelj** [comment]
We 

> > @philippmwirth, the new video interface is much faster when reading sequential frames. That might not work when using multiple workers.
> > How should we change the embedding part of the code? Just switch to 0 or 1 worker in case we can use the new backend?
> 
> I would leave it to the user and document it clearly. IMO it's not a good idea to enforce a certain number of workers.

We can improve the embedding part in another PR. This one would already fix multiple issues we faced when working with videos.

**Review by philippmwirth — APPROVED** [review]

**Pull request merged**